interview-questions.md

```diff
@@ -8,36 +8,243 @@ Then, research the question to expand on your answer. Even if you feel you have
+        end
+- With this method, it is going into the model, Student, and finding whatever is contained in the has with the :id given, which in this case is 2 so it is getting the student with the unique ID of 2
+the students name is Clifford Roecker, and he is the best student of them all :)
 
 action: "GET" location: /students/2/edit
+students#edit-
+-With this method, it is pairing with the update and show methods, it is going into the model, Student, and finding whatever is contained in the hash with the :id given, which in this case is 2 so it is getting the student with the unique ID of 2. it then populates a form with the data already stored for that Student, so the user can edit the info, and then it can utilize the update/patch
 
 action: "PATCH" location: /students/2
+students#update
+        def update
+            student = Student.find(params[:id])
+            student.update(student_params)
+            if student.valid?
+                render json: student
+            else 
+                render json: student.errors
+            end
+        end 
+- with this method, it is using the form that supplied via the GET/EDIT and changing the info about the student with the ID of 2. it is also running it through the student_params, to make sure that it is valid and someone isnt trying to ruin my database of awesome students. 
 
 action: "DELETE" location: /students/2
+students#destroy
+  def destroy
+        animal = Animal.find(params[:id])
+        if animal.destroy
+            render json: animal
+        else 
+            render json: animal.errors
+        end
+    end
+- With this method, it is wow, it is doing the same thing every other method has done to find the student with the id of 2 but this time it is deleting it. 
+If it is able to delete the student located with the unique ID of 2 then it will, if not it will produce an  error... via rendering JSON :)
+
+
+
+CRUD        HTTP            ROUTES
+Create      post            create
+Read        get             edit/new/show/index
+Update      put/patch       update
+Delete      delete          destroy
+
+Prefix        Verb    URL Pattern                  Controller#Action
+students      GET     /students(.:format)          students#index
+              POST    /students(.:format)          students#create
+new_student   GET     /students/new(.:format)      students#new
+edit_student  GET     /students/:id/edit(.:format) students#edit
+student       GET     /students/:id(.:format)      students#show
+              PATCH   /students/:id(.:format)      students#update
+              PUT     /students/:id(.:format)      students#update
+              DELETE  /students/:id(.:format)      students#destroy
+
+
+
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
The idea of user stories is to share the user experience with your application. For instance, 
"I want to have a home page that has a bulletin board with differnt topics/list names"
UX:
"As a user, I can see a bulletin board with different topics/list names on the home page."